src/layers/basic.jl

```diff
@@ -174,27 +174,40 @@ end
 
 """
     Diagonal(α, β)
-    Diagonal(size::Integer...)
+    Diagonal(size::Integer...; bias = true, init = ones32)
 
 Create an element-wise linear layer, which performs
 
     y = α .* x .+ β
 
-The learnable arrays are initialised `α = ones(Float32, size)` and
-`β = zeros(Float32, size)`.
+if `bias` is true, and
+
+    y = α .* x
+
+otherwise. The learnable arrays are initialised `α = ones(Float32, size)` and
+`β = zeros(Float32, size)`. If `init` is specified, the function given to it is 
+called and used to initialise α. The weight matrix and/or the bias vector 
+(with the same size as x) may also be provided explicitly.
 
 Used by [`LayerNorm`](@ref).
 """
-struct Diagonal{T}
-  α::T
-  β::T
+struct Diagonal{A, B}
```

Comment:
In fact, one more suggestion:
```suggestion
struct Diagonal{A<:AbstractArray, B}
```
I think we should either forbid, or explicitly allow,document, and test, the use of a number for the scale. 

Right now you can create such a thing, but there are no tests. And is not handled by `create_bias`:
```
julia> Flux.Diagonal(1.2, 3.4)
Diagonal()

julia> Flux.create_bias(1.2, true, 2,2)
ERROR: MethodError: no method matching create_bias(::Float64, ::Bool, ::Int64, ::Int64)
```

Reply:
In fact the printing lies, too. What `Diagonal()` creates is a zero-dimensional array. 
```
julia> Flux.Diagonal() |> dump
Flux.Diagonal{Array{Float32, 0}}
  α: Array{Float32}(()) fill(1.0f0)
  β: Array{Float32}(()) fill(0.0f0)
```
Which should be fine, it's one scalar parameter. Although probably nothing is tested with this.